.gas-snapshot

```diff
@@ -19,48 +19,48 @@ Certificate_supportsInterface:test() (gas: 5159)
 Certificate_transferFrom:test() (gas: 48618)
 Certificate_transferFrom:test_reverts_when_paused() (gas: 38088)
 Certificate_transferFrom_reverts_ForbiddenTransferAfterMinting:test() (gas: 18960)
-Checkout_buyingFromOneRemoval:test() (gas: 726440)
-Checkout_buyingFromOneRemoval_byApproval:test() (gas: 684688)
-Checkout_buyingFromSingleSupplierWithCustomFee:test() (gas: 547923)
-Checkout_buyingFromTenRemovals:test() (gas: 1976973)
-Checkout_buyingFromTenRemovals_singleSupplier:test() (gas: 1965977)
-Checkout_buyingFromTenRemovals_singleSupplier_byApproval:test() (gas: 1963977)
-Checkout_buyingFromTenRemovals_singleSupplier_withoutFee:test() (gas: 1708460)
-Checkout_buyingFromTenRemovals_withoutFee:test() (gas: 1719464)
-Checkout_buyingFromTenSuppliers:test() (gas: 2855260)
-Checkout_buyingWithAlternateERC20:test() (gas: 544721)
-Checkout_buyingWithAlternateERC20_floatingPointPriceMultiple:test() (gas: 544721)
-Checkout_buyingWithCustomFee:test() (gas: 547446)
-Checkout_swapWithDifferentPermitSignerAndMsgSender:test() (gas: 726890)
+Checkout_buyingFromOneRemoval:test() (gas: 726000)
+Checkout_buyingFromOneRemoval_byApproval:test() (gas: 684424)
+Checkout_buyingFromSingleSupplierWithCustomFeeAndPrice:test() (gas: 553566)
+Checkout_buyingFromTenRemovals:test() (gas: 1975477)
+Checkout_buyingFromTenRemovals_singleSupplier:test() (gas: 1964590)
+Checkout_buyingFromTenRemovals_singleSupplier_byApproval:test() (gas: 1962590)
+Checkout_buyingFromTenRemovals_singleSupplier_withoutFee:test() (gas: 1707603)
+Checkout_buyingFromTenRemovals_withoutFee:test() (gas: 1718584)
+Checkout_buyingFromTenSuppliers:test() (gas: 2853764)
+Checkout_buyingWithAlternateERC20:test() (gas: 544459)
+Checkout_buyingWithAlternateERC20_floatingPointPriceMultiple:test() (gas: 544459)
+Checkout_buyingWithCustomFeeAndPrice:test() (gas: 563667)
```

Comment:
would have been nice to have this test defined as the function name inside the market contract tests. Had to lookup which one of the tests was relevant manually.

diff: +~3%, (at gas price of 100: 0.001622 MATIC, $0.001134). Not as bad as I thought it would be, though still something to call out.